train_general_model.py

```diff
@@ -61,22 +61,11 @@ def main(logger, experiment_config: configs.BaseConfig) -> LightningModule:
 
     train_dataloader, val_dataloader, n_labels = get_dataset_(experiment_config)
 
-    vit = EMGViT(
-        image_size=experiment_config.window_size,
-        patch_size=experiment_config.base_model_config.patch_size,
-        num_classes=n_labels,
-        dim=experiment_config.base_model_config.dim,
-        depth=experiment_config.base_model_config.depth,
-        heads=experiment_config.base_model_config.heads,
-        mlp_dim=experiment_config.base_model_config.mlp_dim,
-        dropout=experiment_config.base_model_config.dropout,
-        emb_dropout=experiment_config.base_model_config.emb_dropout,
-        channels=experiment_config.base_model_config.channels,
-    )
+    base_model = BASE_MODELS[experiment_config.base_model_name](experiment_config)
```

Comment:
create a function make_model(config), by having a  dict we have the definition one jump away anyway

Reply:
the function can be a sequence of ifs or a dict lookup
(i'd rather the ifs)